### PR TITLE
838: primary color styling

### PR DESCRIPTION
**Pull request by c-m-duncan**
There's so many styling changes in here - here's some screenshots. I might have missed some changes

Screen Shot 2022-10-13 at 3 30 39 PM
----------
Screen Shot 2022-10-13 at 3 30 41 PM
----------
Screen Shot 2022-10-13 at 3 30 45 PM
----------
Screen Shot 2022-10-13 at 3 30 50 PM
----------
Screen Shot 2022-10-13 at 3 27 45 PM
----------
Screen Shot 2022-10-13 at 3 27 48 PM
----------
Screen Shot 2022-10-13 at 3 27 51 PM
----------
Screen Shot 2022-10-13 at 3 27 53 PM
----------
Screen Shot 2022-10-13 at 3 30 26 PM
----------
Screen Shot 2022-10-13 at 3 30 54 PM
----------
Screen Shot 2022-10-13 at 3 30 29 PM
----------
Screen Shot 2022-10-13 at 3 31 33 PM
----------
Screen Shot 2022-10-13 at 3 30 33 PM
----------
Screen Shot 2022-10-13 at 3 39 27 PM
----------
Screen Shot 2022-10-13 at 3 30 38 PM
----------
Screen Shot 2022-10-13 at 3 44 27 PM
----------

### CODE REVIEW

File: src/HelmetStyles.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                        color: ${window.PRIMARY_COLOR};
+                    }
+                    login-zone div.user-menu {
+                        background-color: ${window.PRIMARY_COLOR};
+                    }
+                    dataset-detail-view dataset-details .keyword {
+                        color: ${window.PRIMARY_COLOR};
+                    }
+                `}</style>
+            </Helmet>
```
Comment (c-m-duncan) [inline src/HelmetStyles.js:60]: This get's injected in the html <head> allowing us to use these env variables

Comment (bmitchinson) [reply src/HelmetStyles.js:60]: Using helmet is a great creative solution here to link those variables, as mentioned earlier, nice find.

This list of applied styles is a bit long, but I'd much rather have this then attempt to refactor `react_discovery_ui` right now to accommodate a smaller set of selectors. Looks good to me 👍

File: src/components/footer/footer.js [hunk 2]
```diff
@@ -3,15 +3,15 @@ import './footer.scss'
 const Footer = () => {
 
   const links = JSON.parse(window.FOOTER_LINKS),
-    renderLink = ({ url, linkText }) => <a class="link" href={url}> {linkText}</a>;
+    renderLink = ({ url, linkText }) => <a key={linkText} className="link" href={url}> {linkText}</a>;
```
Comment (bmitchinson) [inline src/components/footer/footer.js:6]: This error was drivin me crazy, nice fix!

**Review by bmitchinson — APPROVED** [review]
Approved pending react_discovery_ui version bump

File: package.json [hunk 3]
```diff
@@ -24,8 +24,7 @@
     "@smartcitiesdata/react-discovery-ui": "^2.0.0",
     "react": "~16.8.6",
     "react-dom": "~16.8.6",
-    "react-hamburger-button": "0.0.1",
-    "react-hamburger-menu": "^1.1.1"
```
[the remark this answers is not in the document]
Comment (c-m-duncan) [reply package.json:28]: Do we? I'd expect the syntax of ^2.0.0 to grab the latest 2.X.X release

[the remark this answers is not in the document]
Comment (bmitchinson) [reply package.json:28]: True, maybe it stays as `^2.0.0` here but, at least the package lock will need to be updated. Right now the package.lock strictly says `2.0.0`. Not sure if an `npm i` will update that lock file to include the new `react_discovery_ui` version. Might need to run `npm upgrade @smartcitiesdata/react-discovery-ui` or something, I'll leave that to you though.

**Pull request merged**